app/graphql/users/middlewares.js

```diff
@@ -1,14 +1,50 @@
 const logger = require('../../logger');
+const { user: User } = require('../../models');
+const { EMAIL_REGEX, PASSWORD_REGEX } = require('../../constants');
+const errors = require('../../errors');
 
-const createUser = (resolve, root, args) => {
-  logger.info("Middleware for 'createUser' mutation");
-  // Add different actions that you want to be executed before your resolver, i.e: input validation or caching
-  return resolve(root, args);
+const user = (resolve, root, args) => {
+  logger.info("Middleware for 'user' mutation");
+  const errorList = [];
+  const { email, password } = args.userFields;
+  if (!EMAIL_REGEX.test(email)) {
```

Comment:
In order to keep this method shorter and a bit cleaner, it would be a nice idea to move the code inside both this and the next `if()` to a separate function. 
*HINT*: remember that, in javascript, you can pass functions as parameters.